### PR TITLE
Add `deprecationMessage` for deprecated `DownloadPackageV0` and `NuGet...` tasks

### PR DESCRIPTION
**Pull request by DenisRumyantsev**
***Tasks:***
DownloadPackage@0
NuGetInstaller@0
NuGetPackager@0
NuGetPublisher@0
NuGetRestore@1
NuGet@0

NuGetAuthenticate@0 is not included

***Description:*** Add the tailored deprecation message in addition to the common deprecation message.

***Documentation changes required:*** No

***Added unit tests:*** No

***Attached related issue:*** N/A

***Checklist:***
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by amp-powell — APPROVED** [review]
I'm approving, but possibly we can add links when suggesting the new tasks to use. Also, it's fine to add the depreciationMessage property to DownloadPackageV0, NuGetInstallerV0, and NuGetRestoreV1, but these tasks are already failing.

**Comment by DenisRumyantsev** [comment]
@amp-powell This PR has been created in addition to [another one](https://github.com/microsoft/azure-pipelines-tasks/pull/19443). The link to the documentation is not provided in the tailored deprecation message since it is already included in the common deprecation message.

File: Tasks/DownloadPackageV0/task.json
Comment (geekzter) [inline Tasks/DownloadPackageV0/task.json:1]: This task has already been disabled in Azure DevOps Service, we can still update it though

File: Tasks/NuGetInstallerV0/task.json
Comment (geekzter) [inline Tasks/NuGetInstallerV0/task.json:1]: This task has already been disabled in Azure DevOps Service, we can still update it though

File: Tasks/NuGetInstallerV0/task.loc.json
Comment (geekzter) [inline Tasks/NuGetInstallerV0/task.loc.json:1]: This task has already been disabled in Azure DevOps Service, we can still update it though

**Review by geekzter — CHANGES_REQUESTED** [review]

**Review by geekzter — APPROVED** [review]

**Pull request merged**